extensions/common/crypto/jws2020/src/main/java/org/eclipse/edc/security/signature/jws2020/KeyFactory.java

```diff
@@ -0,0 +1,124 @@
+/*
+ *  Copyright (c) 2023 Bayerische Motoren Werke Aktiengesellschaft (BMW AG)
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Bayerische Motoren Werke Aktiengesellschaft (BMW AG) - initial API and implementation
+ *
+ */
+
+package org.eclipse.edc.security.signature.jws2020;
+
+import com.nimbusds.jose.JOSEException;
+import com.nimbusds.jose.JWSSigner;
+import com.nimbusds.jose.JWSVerifier;
+import com.nimbusds.jose.crypto.ECDSASigner;
+import com.nimbusds.jose.crypto.ECDSAVerifier;
+import com.nimbusds.jose.crypto.Ed25519Signer;
+import com.nimbusds.jose.crypto.Ed25519Verifier;
+import com.nimbusds.jose.crypto.RSASSASigner;
+import com.nimbusds.jose.crypto.RSASSAVerifier;
+import com.nimbusds.jose.jwk.ECKey;
+import com.nimbusds.jose.jwk.JWK;
+import com.nimbusds.jose.jwk.OctetKeyPair;
+import com.nimbusds.jose.jwk.RSAKey;
+
+import java.text.ParseException;
+import java.util.Map;
+import java.util.Objects;
+
+import static java.lang.String.format;
+
+/**
+ * Utility class that has two basic duties:
+ * <ul>
+ *     <li>Create cryptographic keys represented in JWK format ({@link JWK}) out of JSON or JSON-like objects</li>
+ *     <li>Create {@link JWSSigner}s and {@link JWSVerifier}s for any given key represented in {@link JWK} format.</li>
+ * </ul>
+ * <p>
+ * For this, the well-known <a href="https://connect2id.com/products/nimbus-jose-jwt">Nimbus JOSE+JWT library</a> is used.
+ */
+public class KeyFactory {
```

Comment:
For me this class should be located in a more _"core"_ module, as it could (should) also be used in the DID and JWT modules.

Reply:
so maybe it could be in `extensions/common/crypto/crypto-core`?

Reply:
yeap for me `extensions/common/crypto/crypto-core` is a good place